libs/@local/hql/cst/src/expr/signature.rs

```diff
@@ -0,0 +1,153 @@
+use core::{fmt, fmt::Display};
+
+use hql_span::SpanId;
+
+use super::ExprKind;
+use crate::{arena, symbol::Symbol, r#type::Type, Spanned};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct List<'arena, T> {
+    pub items: arena::Box<'arena, [T]>,
+    pub span: SpanId,
+}
+
+impl<'arena, T> List<'arena, T> {
+    #[must_use]
+    pub const fn len(&self) -> usize {
+        self.items.len()
+    }
+
+    #[must_use]
+    pub const fn is_empty(&self) -> bool {
+        self.items.is_empty()
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = &T> {
+        self.items.iter()
+    }
+}
+
+impl<T> Spanned for List<'_, T> {
+    fn span(&self) -> SpanId {
+        self.span
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Signature<'arena> {
+    pub generics: List<'arena, Generic<'arena>>,
+    pub arguments: List<'arena, Argument<'arena>>,
+    pub r#return: Return<'arena>,
+
+    pub span: SpanId,
+}
+
+impl Spanned for Signature<'_> {
+    fn span(&self) -> SpanId {
+        self.span
+    }
+}
+
+impl<'a> Display for Signature<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if !self.generics.is_empty() {
+            f.write_str("<")?;
```

Comment:
```suggestion
            f.write_char('<')?;
```

Reply:
(and other locations)

Reply:
fixed in [4aa946e](https://github.com/hashintel/hash/pull/4828/commits/4aa946e0492aa1387ee8e5afc7a56c300c430755)